### DISCUSSION TITLE
change the duration of animation and update the animation at runtime

### DISCUSSION THREAD

**Discussion by Rabicocoa** [airbnb/lottie-ios#1610] [Q&A]
I need a way to create a custom lottie animation, where I can change the duration of animation and update the animation at runtime. Basically, I want to build a loader using lottie to show the progress animation in case of any kind of downloading.

**Comment by calda** [comment]
If you want to tie the progress of an animation to the progress of a download task, I'd suggest setting `animationView.currentProgress = downloadTask.progress` in your download task's progress handler. If you want to adjust the duration of your animation (e.g. make it play slower or faster) then you could configure `animationView.animationSpeed`.
